### PR TITLE
Improve gold-standard alignment

### PR DESCRIPTION
**Pull request by honnibal**
* Replace the janky alignment data structures with Thinc's `Ragged` array
* For now, use the pytokenizations alignment library. We can probably just implement the function ourselves and remove the dependency, but let's just get this working for now.
* Use the "if all the tags are the same" logic suggested by @svlandeg 
* Use somewhat more permissive NER alignment. Might be bad? Testing.

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tests/test_gold.py [hunk 1]
```diff
@@ -298,7 +296,7 @@ def test_gold_biluo_one_to_many(en_vocab, en_tokenizer):
     gold_words = ["I", "flew", "to", "San", "Francisco", "Valley", "."]
     example = Example.from_dict(doc, {"words": gold_words, "entities": entities})
     ner_tags = example.get_aligned_ner()
-    assert ner_tags == ["O", None, "U-LOC", "O"]
+    assert ner_tags == ["O", "U-ORG", "U-LOC", "O"]
```
Comment (svlandeg) [inline spacy/tests/test_gold.py:300]: The second "word" is "flew to", of which only "flew" is annotated as "ORG", so isn't `None` more appropriate for a mixture of "ORG" and "O" ?

File: spacy/tests/test_gold.py [hunk 2]
```diff
@@ -339,7 +336,7 @@ def test_gold_biluo_misaligned(en_vocab, en_tokenizer):
     gold_words = ["I", "flew to", "San", "Francisco Valley", "."]
     example = Example.from_dict(doc, {"words": gold_words, "entities": entities})
     ner_tags = example.get_aligned_ner()
-    assert ner_tags == [None, None, "B-LOC", "L-LOC", "O"]
+    assert ner_tags == ["B-ORG", "L-ORG", "B-LOC", "L-LOC", "O"]
```
Comment (svlandeg) [inline spacy/tests/test_gold.py:340]: Also here I think the `None`'s are more appropriate as the gold entity can not be mapped back to the original words.

Comment (honnibal) [reply spacy/tests/test_gold.py:340]: Can we update the example so it's more realistic? I keep getting confused by the text

Comment (svlandeg) [reply spacy/tests/test_gold.py:340]: Yea, I know, let me push a quick update.

Comment (svlandeg) [reply spacy/tests/test_gold.py:343]: So basically in `predicted`, the first two tokens are "Mr. and Mrs." and "Smith", but the NER annotation reads "Mrs. Smith". I think the first token should probably be `None`, as "Mr" was never annotated. You could argue about the second token "Smith", whether it should inherit the label from "Mrs. Smith", but I can imagine contexts where half of the entity doesn't make sense to be an entity by itself.

File: spacy/tests/test_gold.py (hunk 1 above)
Comment (svlandeg) [reply spacy/tests/test_gold.py:301]: Updated: the `predicted` token "Mrs. Smith" consists of "Mrs." being annotated as a PERSON and "Smith" as "O", so shouldn't the final result be None instead?

File: spacy/tests/test_gold.py (hunk 2 above)
Comment (honnibal) [reply spacy/tests/test_gold.py:344]: Thanks, I see the problem. I suggested a fix.

We could go a bit further and strip internal whitespace from the spans, to account for stuff like extra newlines. But if the underlying strings match I thing the code is correct.

**Pull request merged**